library/src/main/java/com/google/android/exoplayer2/extractor/flv/AudioTagPayloadReader.java

```diff
@@ -78,8 +97,13 @@ protected boolean parseHeader(ParsableByteArray data) throws UnsupportedFormatEx
   @Override
   protected void parsePayload(ParsableByteArray data, long timeUs) {
     int packetType = data.readUnsignedByte();
-    // Parse sequence header just in case it was not done before.
-    if (packetType == AAC_PACKET_TYPE_SEQUENCE_HEADER && !hasOutputFormat) {
+
+    if (audioFormat != AUDIO_FORMAT_AAC || packetType == AAC_PACKET_TYPE_AAC_RAW) {
```

Comment:
Why have you swapped the if/else order around? It's more natural to have the blocks outputting formats before the ones outputting samples. Can we put them back into that order again?